sql/core/src/main/scala/org/apache/spark/sql/execution/arrow/ArrowConverters.scala

```diff
@@ -71,61 +71,114 @@ private[sql] class ArrowBatchStreamWriter(
 private[sql] object ArrowConverters {
 
   /**
-   * Maps Iterator from InternalRow to serialized ArrowRecordBatches. Limit ArrowRecordBatch size
+   * Maps Iterator from InternalRow to ArrowRecordBatche. Limit ArrowRecordBatch size
    * in a batch by setting maxRecordsPerBatch or use 0 to fully consume rowIter.
    */
-  private[sql] def toBatchIterator(
+  private[sql] def toArrowRecordBatchIterator(
```

Comment:
@BryanCutler and @HyukjinKwon , based on our previous discussion, I moved all arrow related conversions to `ArrowConverters`, so now I added or modified previous c onversions as below 6 functions
* `toArrowRecordBatchIterartor`  :   `Iterator[InternalRow]` -> `Iterator[ArrowRecordBatch]`
* `fromArrowRecordBatchIterartor`  :   `Iterator[ArrowRecordBatch]` -> `Iterator[InternalRow]`
* 
* `toBatchIterator` `Iterator[InteralRow]` -> `Iterator[Array[Byte]]` (will internally call `toArrowRecordBatchIterartor` )
* `fromBatchIterator` `Iterator[Array[Byte]]`-> `Iterator[InteralRow]` (will internally call `fromArrowRecordBatchIterator`)
* 
* `toColumnarBatchIterator` `Iterator[ArrowRecordBatch]` -> `Iterator[ColumnarBatch]` arrow backend
* `fromColumnarBatchIterator` `Iterator[ColumnarBatch]` -> `Iterator[ArrowRecordBatch]`